libcontainer/specconv/spec_linux.go

```diff
@@ -190,6 +195,17 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 		if config.Namespaces.Contains(t) {
 			return nil, fmt.Errorf("malformed spec file: duplicated ns %q", ns)
 		}
+		// if this container is going to have shared cgroup namespace with host,
+		// we just skip the following .Add()
```

Comment:
My point is that you're adding **more** code to special case something that I don't understand why you need to special case it. If you do a `setns(...)` for a namespace that you've already joined then it doesn't matter anyway.

But also I don't understand why you're comparing the link to `/proc/1/ns/cgroup` -- how do you know that runC is inside the same cgroup namespace as the init in it's current PID namespace (that doesn't need to be true)? From what I can see you're going to skip joining cgroups people explicitly asked you to.